hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AssociateTokenRecipientsStep.java

```diff
@@ -181,65 +195,33 @@ private TokenAssociation validateAndBuildAutoAssociation(
             validateFalse(token.hasKycKey(), ACCOUNT_KYC_NOT_GRANTED_FOR_TOKEN);
             validateFalse(token.accountsFrozenByDefault(), ACCOUNT_FROZEN_FOR_TOKEN);
 
-            final var unlimitedAssociationsEnabled =
-                    config.getConfigData(EntitiesConfig.class).unlimitedAutoAssociationsEnabled();
-            if (unlimitedAssociationsEnabled) {
-                dispatchAutoAssociation(token, accountStore, tokenRelStore, context, account);
-                // We still need to return this association to the caller. Since this is used to set in record automatic
-                // associations.
-                return asTokenAssociation(tokenId, accountId);
-            } else {
-                // Once the unlimitedAssociationsEnabled is enabled, this block of code can be removed and
-                // all auto-associations will be done through the synthetic transaction and charged.
-                final var newRelation = autoAssociate(account, token, accountStore, tokenRelStore, config);
-                return asTokenAssociation(newRelation.tokenId(), newRelation.accountId());
+            // We only charge auto-association fees inline if this is a user disaptch; in that case the
+            // contract service will take the auto-association costs from the remaining EVM gas
+            if (context.recordBuilders()
+                    .getOrCreate(SingleTransactionRecordBuilder.class)
```

Comment:
(Gotta say, isn't obvious at this line how a method called `getOrCreate` works given an _interface_ as the target class - had to trace this one down ...)